File: Python/0251-0300/257. Binary Tree Paths.py
# https://leetcode.com/problems/binary-tree-paths/description/

class Solution:
    def binaryTreePaths(self, root: Optional[TreeNode]) -> List[str]:
        result = []

        def getPaths (node, path):
            if node.left == None and node.right == None:
                result.append(path)
                return
            
            # Get path for left and right nodes
            if node.left != None:
                getPaths(node.left, path+"->"+str(node.left.val))
            if node.right != None:
                getPaths(node.right, path+"->"+str(node.right.val))
        
        
        getPaths(root, str(root.val))
        return result

File: Javascript/0251-0300/257. Binary Tree Paths.js
// https://leetcode.com/problems/binary-tree-paths/description/

var binaryTreePaths = function(root) {
    const result = [];

    var getPaths = function(node, path){
        if(node.left == null && node.right == null){
            result.push(path);
            return;
        }

        // Get path for left and right nodes
        if(node.left != null)
            getPaths(node.left, path+"->"+node.left.val);
        if(node.right != null)
            getPaths(node.right, path+"->"+node.right.val);
    };
    
    getPaths(root, root.val+"");
    return result;
};
